webviz_subsurface/plugins/_tornado_plotter_fmu/shared_settings/_selectors.py

```diff
@@ -0,0 +1,49 @@
+from typing import List
+
+import webviz_core_components as wcc
+from dash import Input, Output, callback
+from dash.development.base_component import Component
+from webviz_config.webviz_plugin_subclasses import SettingsGroupABC
+
+from .._plugin_ids import PlugInIDs
+
+
+class Selectors(SettingsGroupABC):
+    """Settings class for labelled dropdown with all response columns"""
+
+    class IDs:
+        # pylint: disable=too-few-public-methods
+        RESPONSE = "response"
+
+    def __init__(
+        self,
+        responses: List[str],
+        initial_response: str,
+    ) -> None:
+        super().__init__("Selectors")
+        self._responses = responses
+        self._initial_response = initial_response
+
+    def layout(self) -> List[Component]:
+        return [
+            wcc.Dropdown(
+                id=self.register_component_unique_id(Selectors.IDs.RESPONSE),
+                label="Response",
+                options=[{"label": i, "value": i} for i in self._responses],
+                value=self._initial_response,
+                multi=False,
+                clearable=False,
+            )
+        ]
+
+    def set_callbacks(self) -> None:
+        @callback(
```

Comment:
This roundtrip is not necessary and should be removed.
I.e. use the component directly in the relevant callback.

Reply:
I agree that you should avoid stores whenever possible due to their performance downsides. Rather implement the callbacks in a manner that all components are connected directly. However, you will need to move your callbacks from the settings group to either the plugin (you could consider if you really need shared settings groups or if it would be better to add the settings groups to your view) or to your view. This has another upside: Without its connections to the plugin's stores, your settings group becomes reusable.